frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java

```diff
@@ -137,18 +159,15 @@ private void doPullNonCurrentBranchFastForwardOnly(Project project,
       @Override
       @UIEffect
       public void onSuccess() {
-        new FetchBackgroundable(
-            project,
-            gitRepository,
-            LOCAL_REPOSITORY_NAME,
-            refspecFromOurRemoteBranchToOurLocalBranch,
-            taskTitle,
-            format(getString("action.GitMachete.BasePullBranchFastForwardOnlyAction.notification.title.pull-fail"),
-                localBranch.getName()),
-            format(getString("action.GitMachete.BasePullBranchFastForwardOnlyAction.notification.title.pull-success"),
-                localBranch.getName()))
-                    .queue();
+        ourRemoteToOurLocalBranchFetchBackgroundable.queue();
       }
-    }.queue();
+    };
+
+    if (FetchAllRemotesAction.isUpToDate()) {
+      remoteRepoToOurRemoteBranchFetchBackgroundable.queue();
```

Comment:
Related: no need to create both bgables since only one is ever executed

Reply:
that ain't true. note that one of the backgroundables executes the other one

Reply:
Whoops right, leave as is then